controller/src/main/java/io/pravega/controller/server/bucket/BucketManager.java

```diff
@@ -140,8 +148,9 @@ protected void doStop() {
         Collection<BucketService> tmp;
         synchronized (lock) { 
             tmp = buckets.values();
+            contributor.close();
```

Comment:
Right, putting the `close` call in the synchronized block is redundant as `close` is already `synchronized` itself. The `close` call should never throw (maybe that's a bold statement) an exception, as its only job is to set an `AtomicBoolean` to true and clear a map (which will never be null). So a `close` call should be an idempotent operation instead of throwing an exception.

 I wanted to call `close` before the other shutdown logic to be conservative with how the health service reports the liveness of the `BucketManager`. I.E., I thought it was better to mark it as `TERMINATED` from the perspective of the health service even though the shutdown may  be in progress, instead of potentially marking it `RUNNING` when the shutdown futures have completed.